notebooks/02-ingestion/10-updating-data.ipynb

```diff
@@ -0,0 +1,1188 @@
+    "PARTITIONED BY DAY'''\n",
+    "\n",
+    "display.run_task(sql)\n",
+    "sql_client.wait_until_ready('example-flights-updates')\n",
+    "display.table('example-flights-updates')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8cdca497-0100-43f6-a64a-1e938fcb09c5",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "## The segment stats like num_rows are populated asynchronously, so you may need \n",
+    "## to run this a few times to see the correct values for each of the segments.\n",
+    "## given that the data is partitioned by day, you will see 30 1-day segments \n",
+    "## spanning 11/1/2005 to 11/30/2005\n",
+    "\n",
+    "sql='''\n",
+    "SELECT \"start\", \"end\", \"num_rows\", \"version\" \n",
+    "FROM sys.segments \n",
+    "WHERE datasource='example-flights-updates'\n",
+    "'''\n",
+    "display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a2457fc9-aca7-4bca-b596-d915460a1d42",
+   "metadata": {},
+   "source": [
+    "## Updating a single row\n",
+    "\n",
+    "\n",
+    "Some time later you are asked to update a row that has an incorrect tail_number the flight was delayed because a plane change was required but this info was updated after the fact.\n",
+    "If you are doing equipment level analysis, you'll need the correct tail_number on each row:\n",
+    "- The flight was incorrectly recorded as using the plane with Tail_Number 'N063AA' \n",
+    "- It actually left late because a change of plane to Tail_Number 'N073AA' \n",
+    "- The key to identify a single row in this table involves, __time (departure time), airline and flight_number\n",
+    "\n",
+    "In a traditional SQL database you would do something like:\n",
+    "```\n",
+    "UPDATE \"example-flights-updates\"\n",
+    "    SET \"tail_number\"='N073AA'\n",
+    "  WHERE \"__time\"= '2005-11-01T00:15:00.000Z' \n",
+    "        AND \"airline\"='AA' \n",
+    "        AND \"flight_number\"=513\n",
+    "```\n",
+    "\n",
+    "Run the following cell to look at that row as it exists in the table before an update:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c0d5f243-63f2-4edb-9a29-2c80ec42f723",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "sql='''\n",
+    "SELECT * FROM \"example-flights-updates\" \n",
+    "WHERE \"__time\"='2005-11-01T00:15:00.000Z' \n",
+    "  AND \"airline\"='AA' \n",
+    "  AND \"flight_number\"=513\n",
+    "'''\n",
+    "display.sql(sql)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a13e5c40-89d1-4f3d-b8f9-f5e3e1f33a3e",
+   "metadata": {},
+   "source": [
+    "### Druid segment overlay logic\n",
+    "At query time Druid selects the segments that are relevant to a query. Typically this involves a set of segment files that cover regular time intervals based on the segment granularity that was ingested. Up to this point you have ingested the data into DAY partitions. \n",
```

Comment:
You can have multiple layered segments that each have different granularity. The latest version for any portion of the time interval is the one that is used for that time interval in particular. It doesn't have an explicit limit but this should not be abused with thousands of 1-second segments in an otherwise hourly or daily segment. I think I need to add some images to explain this better.

Reply:
BTW regular here, just meant periodic. I think its fine. Maybe this is an american vs british english thing ;-) 
DAY partitions is just to indicate that even though we can one partition per day, we can create an update that only requires us to rewrite 1-second of data. So in the first paragraph it is just setting the stage.